### PR TITLE
optimize: adjusting the judgement logic of isTrustedProxy

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?

optimize

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.
- 调整 isTrustedProxy 的判断逻辑，现在的 clientIP 的默认逻辑中 isTrustedProxy 正常情况下一定返回的是 false，导致即便用户的 header 中存在 xr和xl 也不会反回，而是返回 remoteaddress。
- gin 的默认实现是宽松的，isTrustedProxy 会默认返回 true，而不是 false。

### CODE REVIEW

**Comment by li-jin-gou** [comment]
PATL @welkeyever  @Duslia

File: pkg/app/context.go [hunk 1]
```diff
@@ -80,48 +80,72 @@ type ClientIP func(ctx *RequestContext) string
...
+		Mask: net.IPMask{0x0, 0x0, 0x0, 0x0},
+	},
+	{ // ::/0 (IPv6)
+		IP:   net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+		Mask: net.IPMask{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
 	},
 }
 
+var defaultClientIPOptions = ClientIPOptions{
+	RemoteIPHeaders: []string{"X-Forwarded-For", "X-Real-IP"},
```
Comment (welkeyever) [inline pkg/app/context.go:98]: Which one of these two headers has higher priority?

Comment (li-jin-gou) [reply pkg/app/context.go:98]: There are no clear criteria for prioritizing them. X-Forwarded-For is part of RFC 7239 but X-Real-IP is not a standard.  Here is the alignment behavior with the common frames gin and fiber.

**Review by welkeyever — APPROVED** [review]
LGTM

**Review by Duslia — APPROVED** [review]

**Pull request merged**